tgui/packages/tgui/interfaces/Plantmaster.js

```diff
@@ -0,0 +1,414 @@
+            selected={inserted_container !== null}
+            icon="eject"
+            onClick={() => inserted_container !== null ? act('ejectbeaker') : act('insertbeaker')}
+            bold>
+            {inserted_container !== null ? inserted : "Insert Beaker"}
+          </Tabs.Tab>
+        </Tabs>
+
+
+        {category === 'overview' && <PlantOverview cat_lens={category_lengths} container={inserted ? inserted_container : null} seedoutput={seedoutput} />}
+        {category === 'extractables' && <PlantExtractables seedoutput={seedoutput} produce={extractables} sortBy={sortBy} sortAsc={sortAsc} />}
+        {category === 'seedlist' && <PlantSeeds seeds={seeds} seedoutput={seedoutput} splicing={show_splicing} splice_chance={splice_chance} splice_seeds={splice_seeds} sortBy={sortBy} sortAsc={sortAsc} />}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const compare = function (a, b, sortBy, sortAsc) {
+  if (sortBy === "name" || sortBy === "species")
+  {
+    if (sortAsc) {
+      return (''+a[sortBy]).localeCompare(b[sortBy]);
+    }
+    else
+    {
+      return (''+b[sortBy]).localeCompare(a[sortBy]);
+    }
+  }
+  if (sortAsc) {
+    return parseFloat(a[sortBy]) - parseFloat(b[sortBy]);
+  }
+  else
+  {
+    return parseFloat(b[sortBy]) - parseFloat(a[sortBy]);
+  }
+};
+
+const ReagentDisplay = (props, context) => {
+  const { act } = useBackend(context);
+  const container = props.container || NoContainer;
+
+  return (
+    <SectionEx
+      capitalize
+      title={container.name}>
+      {!!props.container || (
+        <Dimmer>
+          <Button
+            icon="eject"
+            fontSize={1.5}
+            onClick={() => act('insertbeaker')}
+            bold>
+            Insert Beaker
+          </Button>
+        </Dimmer>
+      )}
+      <ReagentGraph container={container} />
+      <ReagentList container={container} />
+    </SectionEx>
+  );
+};
+
+const PlantOverview = (props, context) => {
+  const { act } = useBackend(context);
+  const { cat_lens, container } = props;
+  return (
+    <SectionEx
+      capitalize
+      title={"Overview"}
+    >
+      <Flex height="100%" direction="column">
+        <Flex.Item grow>
+          <Box>
+            Items ready for extraction: <b>{cat_lens[0]}</b> <br />
+            Seeds ready for experimentation: <b>{cat_lens[1]}</b>
+          </Box>
+          <ReagentDisplay container={container} />
+        </Flex.Item>
+      </Flex>
+    </SectionEx>
+  );
+};
+const TitleRow = (props, context) => {
+  const { act } = useBackend(context);
+  const { show_damage, sortBy, sortAsc } = props;
+  const headings = ["name", "species", "damage", "genome", "generation", "maturity rate", "production rate", "lifespan", "yield", "potency", "endurance", "controls"];
+  const sortname = ["name", "species", "damage", "genome", "generation", "growtime", "harvesttime", "lifespan", "cropsize", "potency", "endurance", ""];
```

Comment:
These can be constants at the top of the file, don't have to create a new array every time this is rendered